core/jvm/src/main/scala/cats/effect/IOCompanionPlatform.scala

```diff
@@ -43,11 +60,32 @@ private[effect] abstract class IOCompanionPlatform { this: IO.type =>
       Tracing.calculateTracingEvent(fn.getClass))
   }
 
+  /**
+   * Like [[blocking]] but will attempt to abort the blocking operation using thread interrupts
+   * in the event of cancellation. The interrupt will be attempted only once.
+   *
+   * @param thunk
+   *   The side effect which is to be suspended in `IO` and evaluated on a blocking execution
+   *   context
+   *
+   * Implements [[Sync#interruptible]]
+   */
   def interruptible[A](thunk: => A): IO[A] = {
     val fn = Thunk.asFunction0(thunk)
     Blocking(TypeInterruptibleOnce, fn, Tracing.calculateTracingEvent(fn.getClass))
   }
 
+  /**
+   * Like [[blocking]] but will attempt to abort the blocking operation using thread interrupts
+   * in the event of cancellation. The interrupt will be attempted repeatedly until the blocking
+   * operation completes or exits.
```

Comment:
```suggestion
   * in the event of cancelation. The interrupt will be attempted repeatedly until the blocking
   * operation completes or exits.
```

I might also say a few words about *why* this exists. In particular, noting that library authors tend to handle interruption incorrectly, and they may set up a situation where they suppress an interrupt unintentionally (e.g. by catching `Throwable` or even `Exception`) and then resume their work. Repeatedly interrupting is one workaround for this situation, though it does come at a cost, since a *second* blocking thread must be tied down in a hot `while` loop just calling `interrupt()` repeatedly, so the overhead of this effect is relatively high and should only be used in situations where there are no other options.